homeassistant/components/bmw_connected_drive/services.yaml

```diff
@@ -5,45 +5,77 @@
 
 light_flash:
   description: >
-    Flash the lights of the vehicle. The vehicle is identified via the vin
-    (see below).
+    Flash the lights of the vehicle. The vehicle is identified either via its
+    device entry or the VIN. If a VIN is specified, the device entry will be ignored.
   fields:
+    device_id:
```

Comment:
According to the service schema in the code, these fields are required, yet in this service schema in YAML they are not?

Additionally, can't this use regular entity services, and thus, targets?

Reply:
In the service schema, one of `vin` or `device_id` is required. I wasn't able to define this in the yaml (only that _both_ are required).

We do not have regular services entities, as we do not have an entity that would be suited for this. On all cars, we only have a `lock` entity available, but adding these services there doesn't feel right.
We could have `device tracker` entity available on all cars - this would make a little bit more sense, but we would add a device tracker entity that on some cars will never show a location.

If you prefer, I could go for the second option (device tracker for all cars, with some locations being (None, None)) and add the entity services there.